ibm_cloud_sdk_core/token_manager.py

```diff
@@ -0,0 +1,246 @@
+# coding: utf-8
+
+# Copyright 2020 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-many-instance-attributes
+
+
+from abc import ABC
+from threading import Lock
+from typing import Optional
+import time
+import requests
+
+from .api_exception import ApiException
+
+
+class TokenManager(ABC):
+    """An abstract class to contain functionality for parsing, storing, and requesting tokens.
+
+    get_token will retrieve a new token from the url in case the that there is no existing token,
+    or the previous token has expired. Child classes will implement request_token, which will do
+    the actual acquisition of a new token.
+
+    Args:
+        url: The url to request tokens from.
+
+    Keyword Args:
+        disable_ssl_verification: A flag that indicates whether verification of
+            the server's SSL certificate should be disabled or not. Defaults to False.
+        token_name: The key that maps to the token in the dictionary returned from request_token. Defaults to None.
+
+    Attributes:
+        url (str): The url to request tokens from.
+        disable_ssl_verification (bool): A flag that indicates whether verification of
+        the server's SSL certificate should be disabled or not.
+        token_name (str): The key used of the token in the dict returned from request_token.
+        token_info (dict): The most token_response from request_token.
```

Comment:
As was done in https://github.com/IBM/node-sdk-core/pull/89, I'd like you simplify the TokenManager class to only hold the `access_token`, leaving the details of `token_name` and `token_info` to implementing subclasses.

Following the Node implementation, that will mean making `save_token_info` an abstract method that subclasses implement that must set `access_token`, `expire_time`, and `refresh_time`.

This will simplify the implementation and make it more consistent with the Node core.

Reply:
Done in 07136fd